streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractKTableKTableJoinValueGetterSupplier.java

```diff
@@ -30,16 +31,10 @@ public AbstractKTableKTableJoinValueGetterSupplier(final KTableValueGetterSuppli
 
     @Override
     public String[] storeNames() {
-        final String[] storeNames1 = valueGetterSupplier1.storeNames();
-        final String[] storeNames2 = valueGetterSupplier2.storeNames();
-        final ArrayList<String> stores = new ArrayList<>(storeNames1.length + storeNames2.length);
-        for (final String storeName : storeNames1) {
-            stores.add(storeName);
-        }
-        for (final String storeName : storeNames2) {
-            stores.add(storeName);
-        }
-        return stores.toArray(new String[stores.size()]);
+        final ArrayList<String> stores = new ArrayList<>();
+        stores.addAll(Arrays.asList(valueGetterSupplier1.storeNames()));
+        stores.addAll(Arrays.asList(valueGetterSupplier2.storeNames()));
+        return stores.toArray(new String[0]);
```

Comment:
@mjsax @ijuma first of all let me say sorry I am for now having caused 3 people discussing how to concat 2 string arrays :)
seriously though ... since that I already caused this anyway :D ...

I guess I'd rather use the empty string array here, given that I already went for the intentionally short and safe but suboptimal solution. In that case I don't see the point in manually setting up the properly sized array after already forcing extra objects on the GC anyway.
But I agree using the List just to concat arrays is weird (but admittedly failsafe :))